### PR TITLE
feat: add literals for compare method and error handling

### PR DESCRIPTION
**Pull request by taharallouche**
## Context
This, imho, can be a small QoL improvement for the users whose IDEs might list the valid options when entering these arguments. It can also make the code more readable (no need to go through the doc string to figure out the valid options).

Please ignore if it doesn't match the repo's conventions (I don't want to be stepping over any toes here :sweat_smile:). More generally, if you prefer that I open issues to discuss the changes before opening PRs I'd be very happy to do it :pray: 

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: fairlearn/metrics/_disaggregated_result.py [hunk 1]
```diff
@@ -105,7 +108,7 @@ def apply_grouping(
         self,
         grouping_function: str,
         control_feature_names: Optional[List[str]] = None,
-        errors: str = "raise",
+        errors: ErrorHandler = "raise",
```
Comment (adrinjalali) [inline fairlearn/metrics/_disaggregated_result.py:111]:
```suggestion
        errors: "raise" | "coerce" = "raise",
```

**Review by adrinjalali — COMMENTED** [review]
I'd rather list them in every instance, instead of creating a typehint kinda thing. It makes reading the code harder. I'd be happy with the change then.

File: fairlearn/metrics/_disaggregated_result.py (hunk 1 above)
Comment (taharallouche) [reply fairlearn/metrics/_disaggregated_result.py:111]: Hey @adrinjalali , did you mean `errors: Literal["raise, "coerce"] = "raise"` ? I don't think  `errors: "raise" | "coerce" = "raise"` is a valid syntax, is it ? :thinking:

Comment (adrinjalali) [reply fairlearn/metrics/_disaggregated_result.py]: It's valid python syntax, but mypy and pylance complain. So I'm okay with using `Literal` here. pity.

File: fairlearn/metrics/_metric_frame.py [hunk 2]
```diff
@@ -814,7 +814,7 @@ def difference_ci(
         return value
 
     def ratio(
-        self, method: str = "between_groups", errors: str = "coerce"
+        self, method: Literal["between_groups", "to_overall"] = "between_groups", errors: Literal["raise", "coerce"] = "coerce"
```
Comment (adrinjalali) [inline fairlearn/metrics/_metric_frame.py:817]: these lines are too long now. I think `black` should be handling these right.

Comment (taharallouche) [reply fairlearn/metrics/_metric_frame.py]: My bad, I haven't initialized the pre-commit hooks :eyes:

**Comment by adrinjalali** [comment]
@taharallouche note that there's no need to force push here, we squash and merge anyway.

**Review by adrinjalali — APPROVED** [review]

**Pull request merged**